packages/tunnel/src/commands/deploy/index.ts

```diff
@@ -0,0 +1,86 @@
+import { existsSync } from 'node:fs';
+import path from 'node:path';
+
+import { isValidUrl } from '@logto/core-kit';
+import chalk from 'chalk';
+import type { CommandModule } from 'yargs';
+
+import { consoleLog } from '../../utils.js';
+
+import { type DeployCommandArgs } from './types.js';
+import { deployToLogtoCloud } from './utils.js';
+
+const tunnel: CommandModule<unknown, DeployCommandArgs> = {
+  command: ['deploy'],
+  describe: 'Deploy your custom UI assets to Logto Cloud',
+  builder: (yargs) =>
+    yargs.options({
+      auth: {
+        describe: 'Auth credentials of your Logto M2M application. E.g.: <app-id>:<app-secret>',
+        type: 'string',
+      },
+      endpoint: {
+        describe:
+          'Logto endpoint URI that points to your Logto Cloud instance. E.g.: https://<tenant-id>.logto.app/',
+        type: 'string',
+      },
+      path: {
+        alias: ['experience-path'],
+        describe: 'The local folder path of your custom sign-in experience assets.',
+        type: 'string',
+      },
+      resource: {
+        alias: ['management-api-resource'],
+        describe: 'Logto Management API resource indicator. Required if using custom domain.',
+        type: 'string',
+      },
+      verbose: {
+        describe: 'Show verbose output.',
+        type: 'boolean',
+        default: false,
+      },
+    }),
+  handler: async (options) => {
+    const {
+      auth,
+      endpoint,
+      path: experiencePath,
+      resource: managementApiResource,
+      verbose,
+    } = options;
+    if (!auth) {
+      consoleLog.fatal(
+        'Must provide valid Machine-to-Machine (M2M) authentication credentials. E.g. `<app-id>:<app-secret>`'
+      );
+    }
+    if (!endpoint || !isValidUrl(endpoint)) {
+      consoleLog.fatal('A valid Logto endpoint URI must be provided.');
+    }
+    if (!experiencePath) {
+      consoleLog.fatal('A valid experience path must be provided.');
+    }
+    if (!existsSync(path.join(experiencePath, 'index.html'))) {
+      consoleLog.fatal(`The provided experience path must contain an "index.html" file.`);
+    }
+
+    consoleLog.info('⏳ Deploying to Logto Cloud...');
+    await deployToLogtoCloud({ auth, endpoint, experiencePath, managementApiResource, verbose });
+
+    const endpointUrl = new URL(endpoint);
+    consoleLog.info(
+      `🎉 Deployment successful!
+${chalk.green('➜')} You can try your own sign-in UI on Logto Cloud now.
```

Comment:
can we use multiple `consoleLog`s (`consoleLog.plain` in the following lines) for a better shape of code?

Reply:
How about this?

<img width="567" alt="image" src="https://github.com/user-attachments/assets/2d8a6336-22c3-476c-ae55-c5ee7b07d5ac">

Reply:
The check marks were loading spinners when the step was still in progress